### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by stv0g**
I've spent some time auditing the code base for gaining a better understanding.

Here are a few cleanups and refactorings.

The PR includes no functional changes.

### CODE REVIEW

**Comment by stv0g** [comment]
@cnderrauber Do you mind reviewing this PR?

File: candidate_base.go [hunk 1]
```diff
@@ -206,57 +205,53 @@ func (c *candidateBase) start(a *Agent, conn net.PacketConn, initializedCh <-cha
...
+	} else {
+		if a.validateNonSTUNTraffic(c, sAddr) { //nolint:contextcheck
+			// Note: This will return packetio.ErrFull if the buffer ever manages to fill up.
+			if _, err := a.buffer.Write(buf); err != nil {
+				a.log.Warnf("Failed to write packet: %s", err)
+			}
+		} else {
+			a.log.Warnf("Discarded message from %s, not a valid remote candidate", c.addr())
 		}
-
```
Comment (Sean-Der) [inline candidate_base.go:248]: [Effective Go](https://go.dev/doc/effective_go#if) recommends early return over nesting.

If you feel strongly happy to leave! Personally I think it reduces the complexity of reading

File: candidate_base.go [hunk 2]
```diff
@@ -206,57 +205,53 @@ func (c *candidateBase) start(a *Agent, conn net.PacketConn, initializedCh <-cha
...
 	case <-c.closeCh:
 		return
 	}
 
-	log := c.agent().log
-	buffer := make([]byte, receiveMTU)
+	buf := make([]byte, receiveMTU)
 	for {
-		n, srcAddr, err := c.conn.ReadFrom(buffer)
+		n, sAddr, err := c.conn.ReadFrom(buf)
```
Comment (Sean-Der) [inline candidate_base.go:218]: `srcAddr` is kind of common across the Pion code bases. Happy to change if `sAddr` is a common pattern though!


```
sean@SeanLaptop:~/go/src/github.com/pion$ rg -l srcAddr
demo-conference/turn/main.go
webrtc/examples/vnet/show-network-usage/main.go
turn/client_test.go
turn/internal/server/turn_test.go
turn/internal/server/server.go
turn/internal/allocation/five_tuple_test.go
turn/internal/allocation/allocation.go
turn/examples/turn-server/tcp/main.go
turn/examples/turn-server/simple/main.go
turn/examples/turn-server/port-range/main.go
turn/lt_cred.go
turn/examples/turn-server/add-software-attribute/main.go
turn/examples/turn-server/log/main.go
turn/server_config.go
turn/server_test.go
turn/examples/turn-server/tls/main.go
ion-sfu/pkg/sfu/sfu.go
ion-sfu/pkg/sfu/turn.go
ice/connectivity_vnet_test.go
ice/candidate_relay_test.go
ice/candidate_base.go
transport/vnet/net_native_test.go
transport/vnet/conn_test.go
transport/vnet/net_test.go
transport/vnet/conn.go
transport/vnet/nat.go
transport/vnet/chunk.go
```

File: candidate_base.go [hunk 3]
```diff
@@ -206,57 +205,53 @@ func (c *candidateBase) start(a *Agent, conn net.PacketConn, initializedCh <-cha
 }
 
 func (c *candidateBase) recvLoop(initializedCh <-chan struct{}) {
-	defer func() {
-		close(c.closedCh)
-	}()
+	defer close(c.closedCh)
```
Comment (Sean-Der) [inline candidate_base.go:208]: :+1: love cleanup like this :)

**Comment by Sean-Der** [comment]
Really great changes @stv0g. Really inspired by your work on this. Working on older code bases is hard, and you are making it look easy. Great to see the refactoring/typo clean ups.

 In a few cases the changes seem subjective (you have a naming scheme you prefer vs others). Unless we create tooling that makes the new way official I would caution against changes like that.

In the future someone will just come and make the changes back to the way it was before.

**Review by Sean-Der — APPROVED** [review]

**Comment by Sean-Der** [comment]
This is my reasoning around some changes. I don't feel strongly around any of these, but I think worth discussing

```
* raddr -> rAddr - Makes it more readable with casing. Like that change!
* srcAddr -> sAddr - I think it makes the code harder to understand. `src` is more self explanatory then `s`
* buffer -> buff, muxs -> muxes - Neutral change. The churn on the code does make it harder to bisect in the future though.

```

**Comment by stv0g** [comment]
Hi @Sean-Der,

thanks a lot for your quick feedback :) I really appreciate it. I know its a lot of noise for a maintainer of such a big project.

Let me explain my reasoning for some of the changed variable names. Similarly to you, I don't feel strongly about any of those.

However, something I value is consistency which was also one of the main motivation for these changes.
And it triggers my OCD (which is probably a good thing for open-source projects.. So I am happy to tidy up the code-base :D)

> * srcAddr -> sAddr - I think it makes the code harder to understand. `src` is more self explanatory then `s`

Agreed. I will change it to `srcAddr`.

> * raddr -> rAddr - Makes it more readable with casing. Like that change!

To satisfy my demand for consistency, I thought about naming it `remAddr`. However, I disliked it and went for the shorter `rAddr`. Hence, for consistency reasons also for `sAddr`. Yeah, sorry I know its nitpicking. But I am also happy with `srcAddr` / `rAddr`..

> In a few cases the changes seem subjective (you have a naming scheme you prefer vs others).
> Unless we create tooling that makes the new way official I would caution against changes like that.

Yes that is a good point. In my case the tooling is a spell checker extension for VSCode which complains about capitalizations like `portmin` but succeeds to check camel case wording like `portMin`.
We could probably add a spell checker into the CI pipeline, however I think it would produce a lot of false positives and demotivate contributors which are annoyed by failing CI checks.

(E.g. The commit message linter is another example of such an annoyance in my opinion.. Maybe it just takes a bit to get used to it.. But I see almost all first-time contributors stumble over it..)

And another good tool for refactoring code like this is the Go VSCode extension as it makes it really easy to perform project wide refactorings and helps a lot with which code navigation.

File: candidate_base.go (hunk 2 above)
Comment (stv0g) [reply candidate_base.go]: Agreed. I renamed it to `srcAddr` in pion/ice

File: candidate_base.go (hunk 3 above)
Comment (stv0g) [reply candidate_base.go:208]: 😁 I was really wondering how this happened to get into pion/ice originally and questioned my Go skills...

File: candidate_base.go (hunk 1 above)
Comment (stv0g) [reply candidate_base.go:248]: Agreed. I reverted the nested flow.

**Comment by stv0g** [comment]
@Sean-Der I think I addressed all your comments.

Let me know, if you want me to change something else 😃

**Comment by Sean-Der** [comment]
I think this is great, thanks @stv0g !

 Keep these changes coming :) I like the nitpicking, no one wants nits!

**Review by Sean-Der — APPROVED** [review]

**Review by cnderrauber — COMMENTED** [review]
@stv0g Awsome for the cleanup works!! It looks good to me. Left a comment for checking the socket read error.

File: candidate_base.go [hunk 4]
```diff
@@ -206,57 +205,62 @@ func (c *candidateBase) start(a *Agent, conn net.PacketConn, initializedCh <-cha
...
 	}
 
-	log := c.agent().log
-	buffer := make([]byte, receiveMTU)
+	buf := make([]byte, receiveMTU)
 	for {
-		n, srcAddr, err := c.conn.ReadFrom(buffer)
+		n, srcAddr, err := c.conn.ReadFrom(buf)
 		if err != nil {
+			a.log.Warnf("Failed to read from candidate %s: %v", c, err)
```
[the remark this answers is not in the document]
Comment (Sean-Der) [reply candidate_base.go:222]: Nice catch @cnderrauber 

Are you able to grab this, or would you rather I grab it?

[the remark this answers is not in the document]
Comment (stv0g) [reply candidate_base.go:222]: Thanks @cnderrauber 

@Sean-Der I will create a follow-up PR in a minute.

[the remark this answers is not in the document]
Comment (stv0g) [reply candidate_base.go:222]: See #492

**Pull request merged**